lib/ClangImporter/ImportDecl.cpp

```diff
@@ -2041,6 +2041,105 @@ namespace {
       fd->getAttrs().add(new (Impl.SwiftContext) UnsafeAttr(/*Implicit=*/true));
     }
 
+    void generateStaticFactoryForCXXForeignRef(
+        clang::ASTContext &clangCtx, ClassDecl *result,
+        clang::CXXRecordDecl *cxxRecordDecl, clang::Sema &clangSema) {
```

Comment:
Can we move this function to `SwiftDeclSynthesizer.cpp` to keep all the AST synthesis logic in one place?